utils/hotfix_utils/RepoChecker.py

```diff
@@ -0,0 +1,224 @@
+    def get_insts_on_latest_ibex_via_inst_congif(self) -> list:
+        """Get a list of instruments that are on the latest version of IBEX.
+
+        Returns:
+            list: A list of instruments that are on the latest version of IBEX.
+
+        """
+        instrument_list = ChannelAccessUtils().get_inst_list()
+        result_list = []
+        for instrument in instrument_list:
+            if not instrument["seci"]:
+                version = requests.get(
+                    "https://control-svcs.isis.cclrc.ac.uk/git/?p=instconfigs/inst.git;a=blob_plain;f=configurations/config_version.txt;hb=refs/heads/"
+                    + instrument["hostName"]
+                ).text
+                version_first_number = int(version.strip().split(".")[0])
+                if self.debug_mode:
+                    print(
+                        f"DEBUG: Found instrument {instrument['name']} on IBEX version {version_first_number}"
+                    )
+                if (
+                    version_first_number is not None
+                    and version_first_number != "None"
+                    and version_first_number != ""
+                ):
+                    result_list.append(
+                        {
+                            "hostname": instrument["hostName"],
+                            "version": version_first_number,
+                        }
+                    )
+
+        # Get the latest version of IBEX
+        versions = sorted(set([inst["version"] for inst in result_list]))
+        latest_version = versions[-1]
+        second_latest_version = versions[-2]
+
+        # filter out the instruments that are not on the latest version
+        insts_on_latest_ibex = [
+            inst["hostname"]
+            for inst in result_list
+            if (
+                inst["version"] == latest_version
+                or inst["version"] == second_latest_version
+            )
+        ]
+
+        return insts_on_latest_ibex
+
+    def check_instruments(self) -> None:
+        """Run checks on all instruments to find hotfix/changes and log the results.
+
+        Returns:
+            None
+
+        """
+        if self.use_test_inst_list:
+            instrument_list = self.test_inst_list.split(",")
+            instrument_list = [instrument.strip() for instrument in instrument_list]
+            if "" in instrument_list:
+                instrument_list.remove("")
+        else:
+            print("INFO: Getting list of instruments on the 2 latest versions of IBEX")
+            instrument_list = self.get_insts_on_latest_ibex_via_inst_congif()
+
+        instrument_status_lists = {
+            self._uncommitted_changes_key: [],
+            self._undeterminable_at_some_point_key: [],
+            self._commits_on_local_not_upstream_key: [],
+            self._commits_on_upstream_not_local_key: [],
+        }
+
+        # TODO - sort out this shitshow of a loop
```

Comment:
Please can we avoid this kind of code comment generally? There are _many_ other ways to phrase this comment in a more constructive way.

Reply:
Ah yes sorry didn't realise I left that in, I wrote the code a while ago and it needs a refactor but time is moving on fast and other things rely on this to get done so I've left out for now.